File: DSA_Codesheet/Cpp/Selection_Sort.cpp
//Sort any given unsorted array of side n in descending order using the selection sort algorithm.

#include<iostream>
using namespace std;

void selectionSort(int arr[], int n) {
    for(int i=0; i<n-1; i++) {
        int maxIndex = i;
        for(int j=i+1; j<n; j++) {
            if(arr[j]<arr[maxIndex]) {
                maxIndex = j;
            }
        }
        if(maxIndex != i) {
            swap(arr[i], arr[maxIndex]);
        }
    }
}

int main() {
    int n;
    cin >> n;

    int arr[n];

    for(int i=0; i<n; i++) {
        cin >> arr[i];
    }

    selectionSort(arr, n);

    for(int i=0; i<n; i++) {
        cout << arr[i] << " ";
    }

    return 0;
}


/*
Input :

6
2 13 4 1 3 6

Output :

1 2 3 4 6 13

----------------------------------

Input :

5
9 3 6 2 0

Output :

0 2 3 6 9


*/

File: selection_sort.java
public class SelectionSortDescending {
    public static void main(String[] args) {
        int n = 5;
        int[] arr = {8, 6, 2, 5, 1};
        
        selectionSortDescending(arr);
        
        System.out.println("Sorted Array in Descending Order:");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }

    public static void selectionSortDescending(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            int maxIndex = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] > arr[maxIndex]) {
                    maxIndex = j;
                }
            }
            if (maxIndex != i) {
                int temp = arr[i];
                arr[i] = arr[maxIndex];
                arr[maxIndex] = temp;
            }
        }
    }
}
